Write phase 3 articles (#8)
CoherentLabs/StarterGuide: src/content/docs/phase-3-layout-assets-and-styling/Styling & Scalability/handling-aspect-ratios-with-media-queries.mdx

By centering this wrapper, you create an invisible "Safe Zone" that keeps elements anchored to where you place them.

An example implementation of this approach looks like this:

```html title="hud-wrapper.html"
<body>
    <div class="hud-safezone">
        <div class="health-bar">100 HP</div>
        <div class="minimap"></div>
    </div>
</body>
```

```css title="hud-wrapper.css"
.hud-safezone {
    position: relative;
    width: 100vw;
    max-width: 100rem; /* Locks the UI to a specified width */
    height: 100vh;
    margin: 0 auto; /* Centers the container horizontally, creating safe zones on either side on ultrawides */
}

.health-bar {
    position: absolute;
    bottom: 3rem;
    left: 3rem;
}

.minimap {
    position: absolute;
    top: 3rem;
    right: 3rem;
    width: 15rem;
    height: 15rem;
}
```

:::note[The Tradeoffs]
This is structurally robust and highly performant because it avoids media query recalculations entirely. 
It guarantees your HUD never stretches beyond a specific boundary, it's great for simple use cases.

> However... it acts as a global constraint. If a specific element within that wrapper needs to bleed out to the screen edge or be positioned differently, the wrapper restricts it. 

Reviewer: ```suggestion
**However it acts as a global constraint.** If a specific element within that wrapper needs to bleed out to the screen edge or be positioned differently, the wrapper restricts it. 
```

Here, we leave the chat box on the left edge of the screen, but selectively push the minimap inward.

```html title="hud-media.html"
<body>
    <div class="chat-box">System: Welcome...</div>
    <div class="minimap"></div>
</body>
```

```css title="hud-media.css"
/* The Chat Box always stays glued to the extreme left edge for immersion */
.chat-box {
    position: absolute;
    bottom: 2rem;
    left: 2rem;
}

/* 1. Default 16:9 Anchor for the Minimap */
.minimap {
    position: absolute;
    top: 2rem;
    right: 2rem;
    width: 15rem;
    height: 15rem;
}

/* 2. Ultrawide (21:9): Push inward slightly so it doesn't spread out */
@media (min-aspect-ratio: 21/9) {
    .minimap {
        right: 15vw; 
        transform: scale(0.9); /* Shrink slightly so it doesn't block the wider cinematic view */
    }
}

/* 3. Super-Ultrawide (32:9): Push heavily inward into the player's safe zone */
@media (min-aspect-ratio: 32/9) {
    .minimap {
        right: 25vw;
        transform: scale(0.8); /* Shrink so it doesn't block the wider cinematic view */
        opacity: 0.7; /* Lower opacity to reduce peripheral distraction */
    }
}
```

> :::note[The Tradeoffs]
> Allows fine control over every individual element. But managing dozens of media queries for every single UI element can become verbose, hard to maintain and slightly less performant than a pure structural wrapper.
> :::

Reviewer: ```suggestion
:::caution[The Tradeoffs]
Even though media queries allow fine control over every individual element, managing dozens of them can become verbose, hard to maintain and slightly less performant than a pure structural wrapper.
:::
```

> ### Complex UIs use both!

Reviewer: Should there be a `-` in the heading? For example `Complex UIs - use both!`? :D
